pyvisa_sim/component.py

```diff
@@ -82,6 +82,20 @@ def validate_value(self, string_value):
             raise ValueError
         return value
 
+class Connection(object):
+    """A device connection"""
+
+    def __init__(self, q, source_list, function):
+        """
+        :param q: query to use
+        :param source: resource to connect to
+        :param source_parameter: resource output parameter
```

Comment:
The docstring does not match the function signature

Reply:
Also it may be worth to validate that the connection is well formed. But we may need to delay this till after building the whole device to validate the parameters and sources.

Reply:
Fixed.